backend/baseclass.pm

```diff
@@ -1219,21 +1228,72 @@ sub start_ssh_serial {
 
 sub check_ssh_serial {
     my ($self, $fh) = @_;
+    my $read_size = 1024;
+    my $buffer;
 
     if ($self->{serial} && $self->{serial}->sock == $fh) {
-        my $chan = $self->{serial_chan};
-        my $line = <$chan>;
-        if (defined $line) {
-            print $line;
+        if (defined($self->{serial_chan}->read($buffer, $read_size))) {
             open(my $serial, '>>', $self->{serialfile});
-            print $serial $line;
+            print $serial $buffer;
             close($serial);
         }
         return 1;
     }
     return;
 }
 
+=head2 run_ssh_cmd
+
+   $ret = run_ssh_cmd($cmd [, username => ?][, password => ?][,host => ?]);
+   ($ret, $stdout, $stderr) = run_ssh_cmd($cmd [, username => ?][, password => ?][,host => ?], wantarray => 1);
```

Comment:
Maybe we should avoid `wantarray` and create a method called `run_ssh_cmd_with_output()` ?